### ISSUE TITLE
"xgb_model" parameter in train() doesn't increment learning correctly

### ISSUE THREAD

**Issue by AdrianP-** [dmlc/xgboost#2707]
Hi, I was making some test with xgboost and incremental learning and I got a confusing result.  
When I train split dataset, I get different (in accuracy e.g.) result than with whole dataset. 
Example: 

Get data:
```
from numpy import loadtxt
import xgboost as xgb
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

dataset = loadtxt('algorithms/pima-indians-diabetes.data', delimiter=",") #https://archive.ics.uci.edu/ml/machine-learning-databases/pima-indians-diabetes/pima-indians-diabetes.data
X = dataset[:,0:8]
Y = dataset[:,8]
seed = 7
test_size = 0.33
X_train, X_test, y_train, y_test = train_test_split(X, Y, test_size=test_size, random_state=seed)
param = {'max_depth':2, 'eta':1, 'silent':1, 'objective':'binary:logistic' }

xgtrain = xgb.DMatrix(X_train, y_train)
xgtest = xgb.DMatrix(X_test)
```

Run all dataset: 
```
bst = xgb.train(params=param, dtrain=xgtrain)
# make prediction
preds = bst.predict(xgtest)
predictions = [round(value) for value in preds]

# evaluate predictions
accuracy = accuracy_score(y_test, predictions)
print("Accuracy: %.2f%%" % (accuracy * 100.0))
```
--> Accuracy: 75.98%

Run with splitted dataset:
```
init = int(X_train.shape[0] / 2)
xgtrain = xgb.DMatrix(X_train[:init], y_train[:init])
model = xgb.train(params=param, dtrain=xgtrain)

xgtrain = xgb.DMatrix(X_train[init:], y_train[init:])
model = xgb.train(params=param, dtrain=xgtrain, xgb_model=model)

preds = model.predict(xgtest)
predictions = [round(value) for value in preds]

# evaluate predictions
accuracy = accuracy_score(y_test, predictions)
print("Accuracy: %.2f%%" % (accuracy * 100.0))
```
--->Accuracy: 73.62%

What could be happening?

## Environment info
Operating System: Ubuntu 16.04

Package used (python/R/jvm/C++): Python 3.6.1

`xgboost` version used: xgboost-0.6a2

**Comment by AdrianP-** [comment]
I found  this parameters: 
```
{'process_type': 'update',
'updater'     : 'refresh',
'refresh_leaf': True}
```
That it should resolve the problem, however 'process_type' and 'updater' can't create new trees. Why?
